images/pytorch-pods/entrypoint.sh

```diff
@@ -13,24 +13,36 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
 source /setup.sh
 
+set -u
+set -x
 
-# TODO: Make these arguments piped through from jsonnet
-export MACHINE_TYPE=n1-standard-16
-export ACCELERATOR_TYPE=v3-32
-export RUNTIME_VERSION=pytorch-nightly
 export RESOURCE_SUFFIX=$POD_UID
-source /setup-pytorch-pods.sh
+export INSTANCE_TEMPLATE_NAME="instance-template-${RESOURCE_SUFFIX}"
+export TPU_POD_NAME="tpu-pod-${RESOURCE_SUFFIX}"
+export INSTANCE_GROUP_NAME="instance-group-${RESOURCE_SUFFIX}"
 
-set -u
-set -x
+# zone/name -> name
+export TPU_POD_NAME=$(echo ${TPU_NAME} | awk -F'/' '{print $2}')
 
+# HACK: create symlink where other images have `gcloud` installed
+mkdir -p /root/google-cloud-sdk/bin/
+ln -s $(which gcloud) /root/google-cloud-sdk/bin/gcloud
 source /publish.sh
 
-# "$@" should look like python -m torch_xla.distributed.xla_dist --tpu=...
-docker-entrypoint.sh "$@"
-
-# Teardown resources
-/teardown-pytorch-pods.sh
+timeout 180 /setup-instances.sh && \
```

Comment:
If `setup-instances.sh` fails, will the overall Job fail? And would the logs from `entrypoint.sh`, `setup-instances.sh`, and `teardown-instances.sh` appear in stackdriver?

Reply:
Yes and yes. I save the exit code from this chain of commands, unconditionally tear down any resources, and then return that exit code.